2437 instructor thread_id and 2424 voice instructions

cawke opened the pull request:
https://github.com/WEKIT-ECS/MIRAGE-XR/issues/2437
https://github.com/WEKIT-ECS/MIRAGE-XR/issues/2424

The instructor initialization now supports an asynchronous mechanism with `InitResponse`, integrating assistant and thread IDs for AI interactions. Refactored AI-related methods to streamline communication and ensure consistency across various instructor operations. Removed unused packages and updated project configurations for better resource management.

https://github.com/WEKIT-ECS/lib-lee/pull/57

cawke committed:
Add async initialization for instructors and refactor AI queries

The instructor initialization now supports an asynchronous mechanism with `InitResponse`, integrating assistant and thread IDs for AI interactions. Refactored AI-related methods to streamline communication and ensure consistency across various instructor operations. Removed unused packages and updated project configurations for better resource management.

fwild commented on Packages/packages-lock.json:
```diff
@@ -299,51 +299,6 @@
-        "com.unity.ext.flatsharp": "1.1.1",
-        "com.unity.modules.video": "1.0.0",
-        "com.unity.xr.core-utils": "2.3.0",
-        "com.unity.nuget.newtonsoft-json": "3.2.1",
```
Careful - probably better to remove this change to packages-lock.json overall - as we are using the gltfast fix for polyspatial 2.0.4, and as in the past there was a weird downgrade depdency to newtonsoft json (was it for this version or an earlier one?). Anything we break here, will either cause compile errors when compiling for VisionPro, or create broken shaders (again), rendering all gltf 3D objects black.

fwild replied:
to be precise: we use a different gltfast package that requires polyspatial 2.0.4, so in theory the change here should not affect that. But: unless we compiled, we cannot be 100% sure? I tried to google the newtonsoft json dependency, but could not find any info in the package description or changelog (also for the visionos package). Maybe this was resolved by now?

cawke replied:
my mistake, fixed

fwild commented on Packages/manifest.json:
```diff
@@ -59,7 +59,6 @@
     "com.unity.multiplayer.center": "1.0.0",
     "com.unity.nuget.mono-cecil": "1.11.5",
     "com.unity.nuget.newtonsoft-json": "3.2.1",
-    "com.unity.polyspatial.visionos": "2.0.4",
```
see comment below for packages-lock.json

cawke replied:
my mistake, fixed

fwild commented on Assets/MirageXR/ContentTypes/VirtualInstructor/VirtualInstructor.cs:
```diff
@@ -12,6 +13,8 @@ namespace MirageXR
+
         public event Action<AudioClip> OnInstructorResponseAvailable;
         
         private const float CharacterHeight = 1.8f;
```
HistoryFormat below in line 21 is also no longer needed, right?

cawke replied:
yes, fixed

fwild reviewed (approved):
Some small changes needed - I am worried about the package lock update of polyspatial etc.

cawke committed:
Refactor TTS to include voice instructions support

Added support for custom voice instructions in text-to-speech (TTS) methods. Refactored interfaces and implementation to propagate "voiceInstruction" parameter. Updated related classes and UI elements to manage and utilize this feature.

cawke committed:
Removed unused constants and commented code in VirtualInstructor and ContentContainerMobile classes.

cawke committed:
manifest.json, packages-lock.json and ProjectSettings.asset have been reverted

fwild commented:
please update pointer to lib-lee develop (following merge of your PR)

cawke committed:
Merge remote-tracking branch 'origin/develop' into ashubin/2437-instructor-thread_id

# Conflicts:
#	Assets/MirageXR/ContentTypes/VirtualInstructor/Editors/Mobile/AudioStreamPlayer.cs
#	Assets/MirageXR/ContentTypes/VirtualInstructor/Editors/Mobile/ContentContainerMobile.cs
#	Assets/MirageXR/ContentTypes/VirtualInstructor/Editors/Mobile/VirtualInstructorEditor.prefab
#	Assets/MirageXR/ContentTypes/VirtualInstructor/Editors/Mobile/VirtualInstructorViewMobile.cs

cawke committed:
merge
